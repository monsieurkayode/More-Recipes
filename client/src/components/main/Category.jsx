import React, { Component } from 'react';
import { connect } from 'react-redux';
import debounce from 'lodash/debounce';
import PropTypes from 'proptypes';
import { Link } from 'react-router-dom';

import { searchPost } from '../../actions';
import cleanString from '../../../../shared/cleanString';
import Categories from '../../../../shared/categories';
import pascalCase from '../../utils/pascalCase';

/**
 * @summary - Category class declaration
 * @class Category
 * @extends {Component}
 */
export class Category extends Component {
  /**
   * Component constructor
   * @param {object} props
   * @memberOf Category
   */
  constructor(props) {
    super(props);
    this.state = {
      searchType: '',
      searchTerm: ''
    };
  }

  /**
   * @method componentDidMount
   *
   * @param {void} void
   *
   * @returns {void}
   */
  componentDidMount() {
    $((this.search)).on('change', this.handleSearchType);
  }

  /**
   * @method componentWillUpdate
   *
   * @param {void} void
   *
   * @returns {void}
   */
  componentWillUpdate() {
    $('select').material_select();
  }

  /**
   * Handle search
   * @method handleSearch
   *
   * @param {void} void
   *
   * @returns {void}
   */
  handleSearch = () => {
    const { searchType, searchTerm } = this.state;
    if ((cleanString(searchTerm)).length % 3 === 0) {
      this.props.searchPost(searchType, searchTerm);
    }
  }

  /**
   * Handle change
   * @method handleChange
   *
   * @param {object} event
   *
   * @returns {void}
   */
  handleChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  /**
   * Handle search criteria
   * @method handleSearchType
   *
   * @param {object} event
   *
   * @returns {void}
   */
  handleSearchType = (event) => {
    this.setState({
      searchType: event.target.value
    });
  }

  /**
   * @method renderCategory
   *
   * @param {number} index
   *
   * @returns {JSX} JSX
   */
  renderCategory = (index) => {
    const category = Categories[index];
    return (
      <li key={category}>
        <Link
          to={`/category/${category}`}
          className="collapsible-header"
        >
          {pascalCase(category)}
        </Link>
      </li>
    );
  }

  /**
   * Renders the component
   * @method render
   *
   * @returns {JSX} JSX
   */
  render() {
    const handleSearch = debounce(() => { this.handleSearch(); }, 300);
    return (
      <div className="col l2 offset-l1 m2 s12">
        <form id="search-form" onChange={handleSearch}>
          <input
            onChange={this.handleChange}
            value={this.state.searchTerm}
            className="white input-fa"
            name="searchTerm"
            type="text"
            placeholder="&#xf002; Enter search keyword..."
          />
          <select
            ref={(ref) => { this.search = ref; }}
            value={this.state.searchType}
          >
            <option value="" disabled>Search criteria</option>
            <option value="name">Title</option>
            <option value="category">Category</option>
            <option value="ingredients">Ingredients</option>
          </select>
        </form>
        <div className="category hide-on-small-only">
          <div>
            <div>
              <b>Featured Categories</b>
              <p className="divider" />
            </div>
            <ul className="collapsible z-depth-0">
              {Categories.slice(6).map(
                (category, index) => this.renderCategory(index)
              )}
            </ul>
          </div>
        </div>
      </div>
    );
  }
}

Category.propTypes = {
  searchPost: PropTypes.func.isRequired
};

export default connect(null, { searchPost })(Category);
