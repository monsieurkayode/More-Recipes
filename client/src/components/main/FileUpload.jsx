import React from 'react';
import PropTypes from 'proptypes';
/**
 * Hanldes file selection for upload
 *
 * @function handleChange
 *
 * @param {function} handler
 *
 * @returns {object} selected file
 */
const handleChange = handler => ({ target: { files } }) =>
  handler(files.length ? { file: files[0], name: files[0].name } : {});

const FileUpload = ({
  input: { onChange, onBlur, value: omitValue, ...inputProps },
  meta: omitMeta,
  ...props
}) => (
  <div id="upload-btn">
    <label htmlFor="image">
      <input
        type="file"
        id="image"
        onChange={handleChange(onChange)}
        onBlur={handleChange(onBlur)}
        placeholder="Upload Image"
        {...inputProps}
        {...props}
      />
      <span
        className="btn"
      >
        Upload Image
      </span>
    </label>
  </div>
);

FileUpload.propTypes = {
  input: PropTypes.shape({
    onChange: PropTypes.func.isRequired,
    onBlur: PropTypes.func.isRequired
  }).isRequired,
  meta: PropTypes.shape({}).isRequired
};

export default FileUpload;
