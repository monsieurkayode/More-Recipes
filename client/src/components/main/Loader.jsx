import React from 'react';
import PreloaderIcon, { ICON_TYPE } from 'react-preloader-icon';

/**
 * Loader
 * @function Loader
 *
 * @param {void} void
 *
 * @return {JSX} JSX
 */
const Loader = () => (
  <div className="loader">
    <PreloaderIcon
      type={ICON_TYPE.OVAL}
      size={80}
      strokeWidth={7}
      strokeColor="#006664"
      duration={800}
    />
  </div>
);

export default Loader;
