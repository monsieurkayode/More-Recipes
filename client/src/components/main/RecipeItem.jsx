import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'proptypes';
import pascalCase from '../../utils/pascalCase';

/**
 * RecipeItem
 * @function RecipeItem
 *
 * @param {void} void
 *
 * @return {JSX} JSX
 */
const RecipeItem = ({ recipe, upvote, downvote, grid }) =>
  (<div className={`col ${grid}`}>
    <div className="card hoverable views-small">
      <div className="card-image">
        <Link to={`/recipes/${recipe.id}`}>
          <img
            src={recipe.image}
            alt={recipe.recipeName}
          />
        </Link>
        <span className="card-title">{recipe.views} Views</span>
      </div>
      <div
        className="card-content small-cards-home"
      >
        <div className="card-title truncate">
          <strong>{recipe.recipeName}</strong>
        </div>
        <div
          className="card-details"
        >
          <Link
            to={`/category/${recipe.category}`}
            className="chip teal white-text"
          >{pascalCase(recipe.category)}
          </Link>
          <span>
            <div
              onClick={() => downvote(recipe.id)}
              className="unlike right"
            >
              <i
                className="material-icons tiny reaction"
              >thumb_down</i>{recipe.downvote}
            </div>
            <div
              onClick={() => upvote(recipe.id)}
              className="like right"
            >
              <i
                className="material-icons tiny reaction"
              >thumb_up</i>{recipe.upvote}
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>);

RecipeItem.propTypes = {
  recipe: PropTypes.shape({
    id: PropTypes.any,
    views: PropTypes.number,
    upvote: PropTypes.number,
    downvote: PropTypes.number,
    recipeName: PropTypes.string,
    category: PropTypes.string,
    ingredients: PropTypes.string,
    instructions: PropTypes.string,
    image: PropTypes.string
  }).isRequired,
  upvote: PropTypes.func.isRequired,
  downvote: PropTypes.func.isRequired,
  grid: PropTypes.string.isRequired
};

export default RecipeItem;
