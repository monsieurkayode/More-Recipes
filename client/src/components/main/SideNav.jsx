import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'proptypes';
import Avatar from 'react-avatar';

import { logoutAction } from '../../actions';

import dash from '../../../assets/css/img/dash.jpg';


/**
   * SideNav
   * @function SideNav
   *
   * @param {object} props
   *
   * @returns {JSX} JSX
   */
export const SideNav = props => (
  <div>
    <ul className="side-nav teal" id="navlink">
      <li>
        <div className="card-image">
          <img
            className="responsive-img"
            id="dash-img"
            src={dash}
            alt="user-background"
          />
          <div id="side-nav-avatar">
            <Avatar
              className="responsive-img"
              name={props.username}
              size={80}
              color="#345"
              round
            />
          </div>
        </div>
      </li>
      <li>
        <Link className="white-text" to="/recipes/new" >
          <span className="fa fa-lg fa-plus-circle" /> Recipe
        </Link>
      </li>
      <li>
        <Link
          className="white-text"
          to="/dashboard/recipes"
        ><span className="fa fa-lg fa-dashboard" /> Dashboard
        </Link>
      </li>
      <li>
        <a className="white-text" onClick={props.logoutAction}>
          <span className="fa fa-lg fa-sign-out" /> Logout
        </a>
      </li>
    </ul>
  </div>
);

const mapStateToProps = ({ signinState }) => ({
  username: signinState.user.username
});

SideNav.defaultProps = {
  username: ''
};

SideNav.propTypes = {
  logoutAction: PropTypes.func.isRequired,
  username: PropTypes.string,
};

export default connect(mapStateToProps, { logoutAction })(SideNav);
