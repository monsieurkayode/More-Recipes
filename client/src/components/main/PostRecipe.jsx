import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Field, reduxForm } from 'redux-form';
import PropTypes from 'proptypes';

import FileUpload from './FileUpload';
import { Loader } from './';
import { createPost, isFetching } from '../../actions';
import validate from '../../utils/validateRecipe';
import categories from '../../../../shared/categories';
import pascalCase from '../../utils/pascalCase';
import resetPage from '../../utils/resetPage';

/**
 * @summary - PostRecipe class declaration
 * @class PostRecipe
 * @extends {Component}
 */
export class PostRecipe extends Component {
  /**
   * Component constructor
   * @param {object} props
   * @memberOf PostRecipe
   */
  constructor(props) {
    super(props);
    this.state = {
      selectedCategory: 'others'
    };
  }

  /**
   * @method componentDidMount
   *
   * @param {void} void
   *
   * @returns {void}
   */
  componentDidMount() {
    $(('#category')).on('change', this.handleCategory);
  }

  /**
   * @method componentWillUpdate
   *
   * @param {void} void
   *
   * @returns {void}
   */
  componentWillUpdate() {
    $('select').material_select();
  }

  /**
   * Handle submit
   *
   * @method onSubmit
   *
   * @param {void} void
   *
   * @returns {void}
   */
  onSubmit = () => {
    const category = this.state.selectedCategory;
    this.props.isFetching(true, 'PostRecipe');
    const values = this.props.values;
    this.props.createPost(category, values, (message) => {
      resetPage();
      this.props.history.push('/');
      setTimeout(() => Materialize.toast(message, 4000, 'grey darken-2'), 3000);
    });
  }

  /**
   * Handle category selection
   *
   * @method handleCategory
   *
   * @param {object} event
   *
   * @returns {void}
   */
  handleCategory = (event) => {
    this.setState({
      selectedCategory: event.target.value
    });
  }

  /**
   * @method renderInput
   *
   * @param {object} field
   *
   * @returns {JSX} JSX
   */
  renderInput = (field) => {
    const { label, input, meta, type, className, placeholder } = field;
    return (
      <div className="row">
        <div className="input-field col-s12 l12 m12">
          <input
            type={type}
            placeholder={placeholder}
            {...input}
            className={className}
          />
          <label htmlFor className="active">{label}</label>
        </div>
        <div className="red-text right">
          {meta.touched ? meta.error : ''}
        </div>
      </div>
    );
  }

  /**
   * @method renderTextArea
   *
   * @param {object} field
   *
   * @returns {JSX} JSX
   */
  renderTextArea = (field) => {
    const { label, input, meta, type, className, placeholder } = field;
    return (
      <div className="row">
        <div className="input-field col-s12 l12 m12">
          <textarea
            placeholder={placeholder}
            type={type}
            {...input}
            className={className}
          />
          <label htmlFor className="active">{label}</label>
        </div>
        <div className="red-text right">
          {meta.touched ? meta.error : ''}
        </div>
      </div>
    );
  }

  /**
   * @method renderCategory
   *
   * @param {object} field
   *
   * @returns {JSX} JSX
   */
  renderCategory = (field) => {
    const { value } = field;
    return (
      <div className="row">
        <div className="input-field col-s12 l12 m12">
          <select
            id="category"
            value={value}
          >
            <option value="others">Select Category</option>
            {categories
              .map(
                category => (<option value={category} key={category}>
                  {pascalCase(category)}
                </option>)
              )
            }
          </select>
        </div>
      </div>
    );
  }

  /**
   * Renders the component
   * @method render
   *
   * @returns {JSX} JSX
   */
  render() {
    const { handleSubmit, invalid, values, isLoading } = this.props;
    return (
      <div className="container">
        <div className="row">
          { isLoading ?
            <Loader /> :
            <form
              className="col l8 m8 s10 offset-m2 offset-l2 offset-s1"
              onSubmit={handleSubmit(this.onSubmit)}
            >
              <Field
                type="text"
                className=""
                placeholder="Enter recipe name"
                label="Recipe Name"
                name="recipeName"
                component={this.renderInput}
              />

              <Field
                name="category"
                value={this.state.selectedCategory}
                component={this.renderCategory}
              />

              <Field
                type="text"
                placeholder="Markdown supported"
                className="materialize-textarea"
                label="Ingredients"
                name="ingredients"
                component={this.renderTextArea}
              />

              <Field
                type="text"
                placeholder="Markdown supported"
                className="materialize-textarea"
                label="Intsructions"
                name="instructions"
                component={this.renderTextArea}
              />
              <div>
                <Field
                  name="image"
                  component={FileUpload}
                />
                <span className="image-name">
                  <p>
                    {!!values.image && values.image.name}
                  </p>
                </span>
              </div>
              <div className="form-btn-control">
                <Link className="white-text" to="/">
                  <button className="btn red">Cancel</button>
                </Link>
                <button
                  disabled={invalid}
                  type="submit"
                  className="btn"
                >
                  Post
                </button>
              </div>
            </form>}
        </div>
      </div>

    );
  }
}

PostRecipe.defaultProps = {
  values: {},
  errors: {}
};

PostRecipe.propTypes = {
  createPost: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired,
  handleSubmit: PropTypes.func.isRequired,
  invalid: PropTypes.bool.isRequired,
  errors: PropTypes.shape({
    recipeName: PropTypes.string,
    ingredients: PropTypes.string,
    instructions: PropTypes.string
  }),
  values: PropTypes.shape({
    ingredients: PropTypes.string,
    instructions: PropTypes.string
  }),
  isFetching: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired
};

const mapStateToProps = ({ form, isLoading }) => ({
  values: form.PostRecipeForm.values,
  errors: form.PostRecipeForm.syncErrors,
  isLoading: isLoading.postRecipeIsLoading
});

export default reduxForm({
  validate,
  form: 'PostRecipeForm'
})(connect(mapStateToProps, { createPost, isFetching })(PostRecipe));
