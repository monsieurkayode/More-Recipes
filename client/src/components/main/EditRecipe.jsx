import React, { Component } from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form';
import PropTypes from 'proptypes';

import FileUpload from './FileUpload';
import Loader from './Loader';
import { editPost, fetchSingleRecipe, isFetching } from '../../actions';
import validate from '../../utils/validateRecipe';
import categories from '../../../../shared/categories';
import pascalCase from '../../utils/pascalCase';

/**
 * @summary - EditRecipe class declaration
 * @class EditRecipe
 * @extends {Component}
 */
export class EditRecipe extends Component {
  /**
   * Component constructor
   * @param {object} props
   * @memberOf EditRecipe
   */
  constructor(props) {
    super(props);
    this.state = {
      selectedCategory: '',
    };
  }

  /**
   * @method componentDidMount
   *
   * @param {void} void
   *
   * @returns {void}
   */
  componentDidMount() {
    const { recipeId } = this.props.match.params;
    $(('#category')).on('change', this.handleCategory);
    this.props.fetchSingleRecipe(recipeId);
  }

  /**
   * @method componentWillReceiveProps
   *
   * @param {object} nextProps
   *
   * @returns {void}
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.initialValues.category) {
      this.setState({
        selectedCategory: nextProps.initialValues.category
      });
    }
  }

  /**
   * @method componentWillUpdate
   *
   * @param {void} void
   *
   * @returns {void}
   */
  componentWillUpdate() {
    $('select').material_select();
  }

  /**
   * Handle submit
   *
   * @method onSubmit
   *
   * @param {object} values
   * @param {string} category
   *
   * @returns {void}
   */
  onSubmit = (values, category) => {
    const { recipeId } = this.props.match.params;
    category = this.state.selectedCategory;
    this.props.isFetching(true, 'EditRecipe');
    this.props.editPost(recipeId, category, values, (message) => {
      this.props.history.goBack();
      setTimeout(() => Materialize.toast(message, 4000, 'grey darken-2'), 3000);
    });
  }

  /**
   * Handle category selection
   *
   * @method handleCategory
   *
   * @param {object} event
   *
   * @returns {void}
   */
  handleCategory = (event) => {
    this.setState({
      selectedCategory: event.target.value
    });
  }

  /**
   * @method renderInput
   *
   * @param {object} field
   *
   * @returns {JSX} JSX
   */
  renderInput = (field) => {
    const { label, input, meta, type, className, placeholder } = field;
    return (
      <div className="row">
        <div className="input-field col-s12 l12 m12">
          <input
            type={type}
            placeholder={placeholder}
            {...input}
            className={className}
          />
          <label htmlFor className="active">{label}</label>
        </div>
        <div className="red-text right">
          {meta.touched ? meta.error : ''}
        </div>
      </div>
    );
  }

  /**
   * @method renderTextArea
   *
   * @param {object} field
   *
   * @returns {JSX} JSX
   */
  renderTextArea = (field) => {
    const { label, input, meta, type, className, placeholder } = field;
    return (
      <div className="row">
        <div className="input-field col-s12 l12 m12">
          <textarea
            placeholder={placeholder}
            type={type}
            {...input}
            className={className}
          />
          <label htmlFor className="active">{label}</label>
        </div>
        <div className="red-text right">
          {meta.touched ? meta.error : ''}
        </div>
      </div>
    );
  }

  /**
   * @method renderCategory
   *
   * @param {object} field
   *
   * @returns {JSX} JSX
   */
  renderCategory = (field) => {
    const { value } = field;
    return (
      <div className="row">
        <div className="input-field col-s12 l12 m12">
          <select
            id="category"
            value={value}
          >
            <option value={this.state.selectedCategory}>Select Category</option>
            {categories
              .map(
                category => (<option value={category} key={category}>
                  {pascalCase(category)}
                </option>)
              )
            }
          </select>
        </div>
      </div>
    );
  }

  /**
   * Renders the component
   * @method render
   *
   * @returns {JSX} JSX
   */
  render() {
    const { handleSubmit, invalid, initialValues, isLoading } = this.props;
    return (
      <div className="container">
        <div className="row">
          {isLoading ?
            <Loader /> :
            <form
              className="col l8 m8 s10 offset-s1 offset-m2 offset-l2"
              onSubmit={handleSubmit(this.onSubmit)}
            >
              <Field
                type="text"
                className=""
                placeholder="Enter recipe name"
                label="Recipe Name"
                name="recipeName"
                component={this.renderInput}
              />

              <Field
                name="category"
                value={this.state.selectedCategory}
                component={this.renderCategory}
              />

              <Field
                type="text"
                placeholder="Markdown supported"
                className="materialize-textarea"
                label="Ingredients"
                name="ingredients"
                component={this.renderTextArea}
              />

              <Field
                type="text"
                placeholder="Markdown supported"
                className="materialize-textarea"
                label="Intsructions"
                name="instructions"
                component={this.renderTextArea}
              />
              <div>
                <Field
                  name="image"
                  component={FileUpload}
                />
                <span className="image-name">
                  <p>
                    {!!initialValues.image && initialValues.image.name }
                  </p>
                </span>
              </div>

              <div className="form-btn-control">
                <button
                  className="btn red"
                  onClick={() => this.props.history.push('/dashboard/recipes')}
                >
                Cancel
                </button>
                <button
                  disabled={invalid}
                  type="submit"
                  className="btn"
                >
                Update
                </button>
              </div>
            </form>}
        </div>
      </div>
    );
  }
}

EditRecipe.propTypes = {
  editPost: PropTypes.func.isRequired,
  fetchSingleRecipe: PropTypes.func.isRequired,
  initialValues: PropTypes.shape({
    id: PropTypes.number,
    views: PropTypes.number,
    upvote: PropTypes.number,
    downvote: PropTypes.number,
    recipeName: PropTypes.string,
    category: PropTypes.string,
    ingredients: PropTypes.string,
    instructions: PropTypes.string,
    image: PropTypes.shape({})
  }).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
    goBack: PropTypes.func.isRequired,
  }).isRequired,
  handleSubmit: PropTypes.func.isRequired,
  invalid: PropTypes.bool.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      recipeId: PropTypes.string.isRequired
    }).isRequired
  }).isRequired,
  isLoading: PropTypes.bool.isRequired,
  isFetching: PropTypes.func.isRequired
};

const mapStateToProps = ({ currentRecipe, isLoading }) => ({
  initialValues: {
    recipeName: currentRecipe.recipeName,
    category: currentRecipe.category,
    ingredients: currentRecipe.ingredients,
    instructions: currentRecipe.instructions,
    image: {
      file: 'empty',
      name: currentRecipe.image
    }
  },
  isLoading: isLoading.editRecipeIsLoading
});

export default compose(
  connect(mapStateToProps, { editPost, fetchSingleRecipe, isFetching }),
  reduxForm({
    validate,
    form: 'EditRecipeForm',
    enableReinitialize: true,
  }),
)(EditRecipe);
