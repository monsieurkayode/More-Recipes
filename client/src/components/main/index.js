/* jshint esversion: 6 */
import Category from './Category';
import Recipes from './Recipes';
import RecipeItem from './RecipeItem';
import TopRecipeItem from './TopRecipeItem';
import TopRecipes from './TopRecipes';
import SignupForm from './SignupForm';
import SigninForm from './SigninForm';
import TextField from './TextField';
import PostRecipe from './PostRecipe';
import EditRecipe from './EditRecipe';
import SideNav from './SideNav';
import WelcomeMessage from './WelcomeMessage';
import Loader from './Loader';

export {
  Category,
  Recipes,
  RecipeItem,
  TopRecipeItem,
  TopRecipes,
  PostRecipe,
  EditRecipe,
  SignupForm,
  SigninForm,
  TextField,
  SideNav,
  WelcomeMessage,
  Loader
};
