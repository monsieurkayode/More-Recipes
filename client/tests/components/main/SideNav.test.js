/* eslint-disable no-undef */
import React from 'react';

import { SideNav } from '../../../src/components/main/SideNav';

const setup = () => {
  const props = {
    username: 'username'
  };

  const wrapper = shallow(<SideNav {...props} />);

  return { wrapper };
};

const { wrapper } = setup();

describe('SideNav', () => {
  it('should render correctly',
    (done) => {
      expect(toJson(wrapper)).toMatchSnapshot();
      done();
    });
});
