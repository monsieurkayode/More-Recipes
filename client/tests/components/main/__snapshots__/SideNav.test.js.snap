// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav should render correctly 1`] = `
<div>
  <ul
    className="side-nav teal"
    id="navlink"
  >
    <li>
      <div
        className="card-image"
      >
        <img
          alt="user-background"
          className="responsive-img"
          id="dash-img"
          src="test-file-stub"
        />
        <div
          id="side-nav-avatar"
        >
          <Avatar
            className="responsive-img"
            color="#345"
            email={null}
            facebookId={null}
            fgColor="#FFF"
            googleId={null}
            maxInitials={null}
            md5Email={null}
            name="username"
            round={true}
            size={80}
            skypeId={null}
            style={null}
            textSizeRatio={3}
            twitterHandle={null}
            unstyled={false}
            value={null}
            vkontakteId={null}
          />
        </div>
      </div>
    </li>
    <li>
      <Link
        className="white-text"
        replace={false}
        to="/recipes/new"
      >
        <span
          className="fa fa-lg fa-plus-circle"
        />
         Recipe
      </Link>
    </li>
    <li>
      <Link
        className="white-text"
        replace={false}
        to="/dashboard/recipes"
      >
        <span
          className="fa fa-lg fa-dashboard"
        />
         Dashboard
      </Link>
    </li>
    <li>
      <a
        className="white-text"
      >
        <span
          className="fa fa-lg fa-sign-out"
        />
         Logout
      </a>
    </li>
  </ul>
</div>
`;
